### PR TITLE
FAI-12648: Task board relationship hotspot

### PR DESCRIPTION
**Pull request by ted-faros**
## Description

Dynamically disable model resets for jira source

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### CODE REVIEW

**Review by chalenge — COMMENTED** [review]
looks fine to me, will defer to @cjwooo @ypc-faros for approval

File: faros-airbyte-cdk/src/sources/source.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ export abstract class AirbyteSource<
    */
   abstract discover(config: Config): Promise<AirbyteCatalogMessage>;
 
+  /**
+   * Override this method to update the config before running the connector.
+   * The returned config will be passed to the source and destination.
+   */
+  async onBeforeRun(config: Config): Promise<Config> {
```
Comment (ypc-faros) [inline faros-airbyte-cdk/src/sources/source.ts:29]: Wasn't it possible to use [onBeforeRead](https://github.com/faros-ai/airbyte-connectors/blob/a17701280ff6031f3dacc182bcc94e516183209e/faros-airbyte-cdk/src/sources/source.ts#L32) ?

File: destinations/airbyte-faros-destination/src/converters/converter.ts [hunk 2]
```diff
@@ -85,6 +85,7 @@ export class StreamContext {
   private readonly streamsByResetModel: Dictionary<Set<string>> = {};
 
   private readonly resetModels: Set<string> = new Set();
+  private readonly noResetModels: Set<string> = new Set();
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/converter.ts:88]: nit: Maybe `skipResetModels` ?

File: faros-airbyte-cdk/src/sources/source.ts (hunk 1 above)
Comment (cjwooo) [reply faros-airbyte-cdk/src/sources/source.ts:29]: onBeforeRead is called after the source has already sent its config to the destination.

File: destinations/airbyte-faros-destination/src/converters/converter.ts (hunk 2 above)
Comment (cjwooo) [reply destinations/airbyte-faros-destination/src/converters/converter.ts:88]: +1

**Review by cjwooo — APPROVED** [review]

**Review by ypc-faros — APPROVED** [review]
🚀

**Pull request merged**